docs/_templates/module.rst

```diff
@@ -0,0 +1,36 @@
+:html_theme.sidebar_secondary.remove:
+{{ fullname | escape | underline}}
+
+.. autoclass:: {{ fullname }}
+   :members:
+   :show-inheritance:
+   :undoc-members:
+   :member-order: bysource
+
+   {% block attributes %}
+   {% if attributes %}
+   .. rubric:: Attributes
+
+   .. autosummary::
+      :toctree:
+      {% for item in attributes %}
+      {% if item not in inherited_members %}
```

Comment:
Yes. This is the first-pass template implementation that will be extending. Fundamentally, all inherited attribtues are accessible through the inherited classes "Base" at the top of each class. 

There are a few caveats:
* There is an issue of overloading inherited attributes, sometimes a single attribute is declared three times particularly for the mediums classes, this makes linking or generating automatic inherited attributes slightly inaccurate.

We have a few options I have been working on/thinking about:
* We can generate everything and it becomes as overloaded as the current old docs are per class
* We can just generate class specific attributes and methods, and create links to each parent class so they can explore parent attributes. It would be pretty nice if still we can generate just a table of inherited attribute links to the parent class geneated docstrings, but the issue is the overloaded attributes that complicate this. This is my preferred option and what I've been working on. For now, technically all the documentation is complete as all the parent classes are generated.

Reply:
your approach sounds good. thanks for explaining